awscli/customizations/cloudfront.py

```diff
@@ -50,14 +86,83 @@ def __init__(self):
 
     def add_to_params(self, parameters, value):
         if value is not None:
-            caller_reference = 'cli-%s-%s' % (
-                int(time.time()), random.randint(1, 1000000))
             parameters['InvalidationBatch'] = {
-                "CallerReference": caller_reference,
+                "CallerReference": unique_string(),
                 "Paths": {"Quantity": len(value), "Items": value},
                 }
 
 
+def _add_exclusive_arguments(argument_table, args,
+                             exclusive_to='distribution-config', **kwargs):
+    argument_table[exclusive_to].required = False
+    for arg in args:
+        argument_table[arg.name] = arg
+
+
+class ExclusiveArgument(CustomArgument):
+    DOC = '%s This argument and --%s are mututally exclusive.'
+
+    def __init__(self, name, exclusive_to, help_text=''):
+        super(ExclusiveArgument, self).__init__(
+            name, help_text=self.DOC % (help_text, exclusive_to))
+
+    def add_to_params(self, parameters, value):
```

Comment:
See your original comment you made. We are a little out of sync here, but I like your suggestion.